bulk/detail/cuda_launcher/triple_chevron_launcher.hpp

```diff
@@ -124,9 +124,13 @@ class triple_chevron_launcher : protected triple_chevron_launcher_base<block_siz
         {
 #if __BULK_HAS_CUDART__
 #  ifndef __CUDA_ARCH__
-          cudaConfigureCall(dim3(num_blocks), dim3(block_size), num_dynamic_smem_bytes, stream);
-          cudaSetupArgument(task, 0);
-          bulk::detail::throw_on_error(cudaLaunch(super_t::global_function_pointer()), "after cudaLaunch in triple_chevron_launcher::launch()");
+          // create an array of void*s which point to each argument
+          void* pointers_to_args[] = {reinterpret_cast<void*>(const_cast<task_type*>(&task))};
+
+          bulk::detail::throw_on_error(
+            cudaLaunchKernel(super_t::global_function_pointer(), dim3(num_blocks), dim3(block_size), pointers_to_args, num_dynamic_smem_bytes, stream),
```

Comment:
Needs to add a `reinterpret_cast<void*>(...)`  to avoid the error: 
```
/usr/local/cuda-10.1/include/cuda_runtime.h:200:1: note: candidate: ‘template<class T> cudaError_t cudaLaunchKernel(const T*, dim3, dim3, void**, size_t, cudaStream_t)’
 static __inline__ __host__ cudaError_t cudaLaunchKernel(
 ^~~~~~~~~~~~~~~~
/usr/local/cuda-10.1/include/cuda_runtime.h:200:1: note:   template argument deduction/substitution failed:
/home/augalves/Development/Release/Hydra/hydra/detail/external/thrust/system/cuda/detail/bulk/detail/cuda_launcher/triple_chevron_launcher.hpp:132:40: note:   types ‘const T’ and ‘void( _gigantic template instantion_)’ have incompatible cv-qualifiers
```

So the code should become :

```
 bulk::detail::throw_on_error(
            cudaLaunchKernel(reinterpret_cast<void*>(super_t::global_function_pointer()), dim3(num_blocks), dim3(block_size), pointers_to_args, num_dynamic_smem_bytes, stream),
            "after cudaLaunchKernel in triple_chevron_launcher::launch()"
          );
```